modyn/evaluator/internal/metrics/f1_score.py

```diff
@@ -0,0 +1,104 @@
+from enum import Enum
+from typing import Any
+
+import numpy as np
+import torch
+from modyn.evaluator.internal.metrics.abstract_decomposable_metric import AbstractDecomposableMetric
+
+
+class F1ScoreTypes(Enum):
+    MACRO = "macro"
+    MICRO = "micro"
+    WEIGHTED = "weighted"
+    BINARY = "binary"
+
+
+VALID_SCORE_TYPES = {score_type.value for score_type in F1ScoreTypes}
+
+
+class F1Score(AbstractDecomposableMetric):
+    """
+    F1-score implementation.
+    """
+
+    def __init__(self, evaluation_transform_func: str, config: dict[str, Any]) -> None:
+        super().__init__(evaluation_transform_func, config)
+
+        if "num_classes" not in config:
+            raise ValueError("Must provide num_classes to the F1-score metric.")
+        self.num_classes = config["num_classes"]
+
+        self.average = F1ScoreTypes.MACRO
+        if "average" in config:
+            average_type_name = config["average"]
+            if average_type_name in VALID_SCORE_TYPES:
+                self.average = F1ScoreTypes(average_type_name)
+            else:
+                raise ValueError(
+                    f"Provided invalid average strategy {average_type_name}. "
+                    f"Must be one of: {', '.join(VALID_SCORE_TYPES)}"
+                )
+
+        self.pos_label = 1
+        if "pos_label" in config:
+            self.pos_label = config["pos_label"]
```

Comment:
I am probably just blind, but is there - pipeline schema, comment, ... - an explanation of what `pos_label` is referring to? If not, maybe we could add a small comment here

Reply:
this might be still open

Reply:
(feel free to resolve this comment and merge by yourself)